Library/Homebrew/test/test_ENV.rb

```diff
@@ -141,4 +141,32 @@ def test_initializes_deps
     assert_equal [], @env.deps
     assert_equal [], @env.keg_only_deps
   end
+
+  def test_unsupported_cxx11
+    ["gcc", "gcc-4.7"].each do |compiler|
+      @env["HOMEBREW_CC"] = compiler
+      assert_raises do
+        @env.cxx11
+      end
+      assert_nil @env["HOMEBREW_CCCFG"]
```

Comment:
I would prefer if we didn't make more assumptions about `HOMEBREW_CCCFG` than strictly necessary. (Our build defaults unrelated to C++11 might change at some point.) Thus, I'd modify this to check that there is no `x` in the environment variable:

```ruby
refute_match "x", @env["HOMEBREW_CCCFG"]
```